### PR TITLE
New regex_subset_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #962

### CODE REVIEW

File: R/regex_subset_linter.R [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      "not(parent::*[LEFT_ASSIGN or EQ_ASSIGN or RIGHT_ASSIGN])"
+    )
+    # However the equality operator in XPath works for expr[.] = expr/expr[.],
+    #   i.e., node1 = node2, suits our purpose -- we need that the
+    #   xml2::xml_text() that would come out from the matched nodes is the same,
+    #   i.e. that whatever expression comes in <expr>[grepl(pattern, <expr>)]
+    #   matches exactly, e.g. names(x)[grepl(pattern, names(x))].
+    subset_cond_fmt <- xp_and(
+      "expr[SYMBOL_FUNCTION_CALL[%s]]",
+      "expr[position() = %d] = parent::expr/expr[1]"
```
Comment (AshesITR) [inline R/regex_subset_linter.R:46]: Cool! What is the XPath definition of `=` in this context?
It must ignore attributes I see (col1 e.g. will differ), anything else?

File: R/regex_subset_linter.R [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+regex_subset_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$xml_parsed_content) == 0L) {
+      return(list())
+    }
+
+    xml <- source_file$xml_parsed_content
+
+    parent_expr_cond <- xp_and(
+      "OP-LEFT-BRACKET",
```
Comment (AshesITR) [inline R/regex_subset_linter.R:34]: Is `LBB` also worth linting?

**Review by AshesITR — DISMISSED** [review]

File: R/regex_subset_linter.R (hunk 1 above)
Comment (MichaelChirico) [reply R/regex_subset_linter.R:46]: I genuinely don't know. would be great to find documentation on this... thus far it's "just worked" wherever needed though (also tested in a few other linters, not sure if any are included yet)

Comment (MichaelChirico) [reply R/regex_subset_linter.R:46]: Thanks for the spur to check again... found some docs!

https://www.w3.org/TR/1999/REC-xpath-19991116/#booleans

Courtesy this SO Q&A: https://stackoverflow.com/a/4629660/3576984

Comment (MichaelChirico) [reply R/regex_subset_linter.R:45]: Added that link to the comment

**Review by AshesITR — APPROVED** [review]

**Pull request merged**